transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/scheduling/persistence/SQLiteEventStore.java

```diff
@@ -0,0 +1,443 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.android.datatransport.runtime.scheduling.persistence;
+
+import android.content.ContentValues;
+import android.content.Context;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteDatabaseLockedException;
+import android.database.sqlite.SQLiteOpenHelper;
+import android.os.Build;
+import android.os.SystemClock;
+import android.support.annotation.Nullable;
+import android.support.annotation.RestrictTo;
+import android.support.annotation.VisibleForTesting;
+import android.support.annotation.WorkerThread;
+import com.google.android.datatransport.Priority;
+import com.google.android.datatransport.runtime.EventInternal;
+import com.google.android.datatransport.runtime.synchronization.SynchronizationException;
+import com.google.android.datatransport.runtime.synchronization.SynchronizationGuard;
+import com.google.android.datatransport.runtime.time.Clock;
+import com.google.android.datatransport.runtime.time.Monotonic;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+import java.util.Set;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/** {@link EventStore} implementation backed by a SQLite database. */
+@Singleton
+@WorkerThread
+public class SQLiteEventStore implements EventStore, SynchronizationGuard {
```

Comment:
Chatted with @vkryachko offline to determine if Room is a good fit for us

Reply:
**Ashwin**: Did you consider using Room ?

**Vlad**: yep, it transitively pulls hundreds of KB of dependencies
I essentially decided to punt this decision until/if we have a sofisticated enough setup(now it is just 3 simple tables)

**Ashwin**: Just tried it on my app. I do see it adding 100K without proguard. The interesting thing is that clearcut adds more than this (for many products, but not for some others since there are common transitive deps). So the overall could be a net gain

**Vlad**: well, we are a bit over 200KB ourselves already and we are not feature complete yet, so I would not add another 100 lightly

**Vlad**: (that size includes our transitive deps of course)